var/spack/repos/builtin/packages/raja/package.py

```diff
@@ -22,17 +22,87 @@
 # License along with this program; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 ##############################################################################
+#
+# This is a template package file for Spack.  We've put "FIXME"
+# next to all the things you'll want to change. Once you've handled
+# them, you can save this file and test your package like this:
+#
+#     spack install raja
+#
+# You can edit this file again by typing:
+#
+#     spack edit raja
+#
+# See the Spack documentation for more information on packaging.
+# If you submit this package back to Spack as a pull request,
+# please first remove this boilerplate and all FIXME comments.
+#
 from spack import *
 
 
-class Raja(Package):
-    """RAJA Parallel Framework."""
-    homepage = "http://software.llnl.gov/RAJA/"
+class Raja(CMakePackage):
+    """A performance portable parallel programming framework with focus on ease of development"""
+
+    # FIXME: Add a proper url for your package's homepage here.
+    homepage = "http://github.com/LLNL/RAJA"
+    url      = "https://github.com/LLNL/RAJA/archive/v0.2.5.zip"
 
-    version('git', git='https://github.com/LLNL/RAJA.git', branch="master")
+    version('0.2.5', 'ec326e2938d1505078aa60cdfeaf48d0')
+    version('0.2.4', '9facff6cb5bc163fda147f9718fc34f6')
+    version('0.2.3', 'dabfcf6c95867dc0c01bae0d7ba3b964')
+    version('0.2.2', '60835e76094751d8719311bc4236ce4d')
+    version('0.2.1', '35741363f8b912c7f11685602a568d0f')
+    version('0.2.0', 'ed076fafc8cfa03f4badec10dee729fb')
+    version('0.1.0', 'a05098246c9dea465bcfd46b1268e517')
 
-    def install(self, spec, prefix):
-        with working_dir('build', create=True):
-            cmake('..', *std_cmake_args)
-            make()
-            make('install')
+    variant('RAJA_ENABLE_EXAMPLES', default ="ON")
+    variant('RAJA_REPORT_FT', default ="OFF")
+    variant('RAJA_CUDA_ARCH', default ="sm_35")
+    variant('RAJA_USE_GETTIME', default ="OFF")
```

Comment:
It seems we can potentially create *a lot* of variants automatically that are not adhering to the style rules we decided a while ago:

1. Every variant is capital, and I may remember it wrong, but I think we decided to have all small letters.
1. "OFF" and "ON" should appear as `False` and `True` in variants, at least imo
1. The prefix `RAJA_` makes sense within a `CMakeLists.txt`, but here is redundant. I think we should remove it

What I am suggesting are not radical changes, just a bit of processing to turn:
```python
variant('RAJA_ENABLE_OPENMP', default ="ON")
```
into
```python
variant('openmp', default=True)  # RAJA_ENABLE_OPENMP
```